connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractWorkerSourceTask.java

```diff
@@ -458,13 +458,8 @@ boolean sendRecords() {
     }
 
     protected List<SourceRecord> poll() throws InterruptedException {
-        try {
-            return task.poll();
-        } catch (RetriableException | org.apache.kafka.common.errors.RetriableException e) {
-            log.warn("{} failed to poll records from SourceTask. Will retry operation.", this, e);
-            // Do nothing. Let the framework poll whenever it's ready.
-            return null;
-        }
+        retryWithToleranceOperator.reset();
+        return retryWithToleranceOperator.execute(task::poll, Stage.TASK_POLL, this.getClass());
```

Comment:
Looks like this currently only handles `org.apache.kafka.connect.errors.RetriableException`, should we also add `org.apache.kafka.common.errors.RetriableException`? Looks like there's some history here - https://github.com/apache/kafka/pull/6675

Reply:
Oh.. I just went by what the KIP says in the Proposed Approach section. I would assume this change would need an amendment of the KIP .